tests/unit/Tools/TaskRateCalculatorTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C\UnitTest\Tools;
+
+use StoreKeeper\WooCommerce\B2C\Models\TaskModel;
+use StoreKeeper\WooCommerce\B2C\Tools\TaskHandler;
+use StoreKeeper\WooCommerce\B2C\Tools\TaskRateCalculator;
+use StoreKeeper\WooCommerce\B2C\UnitTest\AbstractTest;
+
+class TaskRateCalculatorTest extends AbstractTest
+{
+    public function testEmptyTasks()
+    {
+        $now = '1970-01-01 02:00:00';
+        $calculator = new TaskRateCalculator([], $now, 60); // Ensure 60 minutes/1 hour range
+        $incomingRate = $calculator->calculateIncoming();
+        $processedRate = $calculator->calculateProcessed();
+
+        $this->assertEquals(0, $incomingRate, 'Should not return error but 0');
+        $this->assertEquals(0, $processedRate, 'Should not return error but 0');
+    }
+
+    public function testTaskIncomingRate()
+    {
+        // make a task
+        $this->createTaskWithCreatedDate(1, '1970-01-01 01:30:00');
+        $now = '1970-01-01 02:00:00';
+        $hourAgo = date('Y-m-d H:i:s', strtotime("{$now} -1 hours"));
+        $tasks = TaskModel::getTasksByCreatedDateTimeRange($hourAgo, $now, 0, 'ASC');
+
+        $calculator = new TaskRateCalculator($tasks, $now, 60); // Ensure 60 minutes/1 hour range
+        $incomingRate = $calculator->calculateIncoming();
+
+        $this->assertEquals(2.0, $incomingRate, 'Expected 2 incoming rate per hour for 1 task every 30 minutes');
```

Comment:
Looking at this test, the way of calculating incoming rate is not correct, because it should show 1. incoming rate is just the amount of tasks created in last hour.

Reply:
Changed implementation to just count tasks in the previous hour.